src/coreclr/jit/hwintrinsicxarch.cpp

```diff
@@ -1024,7 +1024,7 @@ GenTree* Compiler::impBaseIntrinsic(NamedIntrinsic        intrinsic,
         {
             assert(sig->numArgs == 2);
 
-            if ((simdSize != 32) || varTypeIsFloating(simdBaseType) || compExactlyDependsOn(InstructionSet_AVX2))
+            if ((simdSize != 32) || compExactlyDependsOn(InstructionSet_AVX2))
```

Comment:
I was originally going to handle this differently but this likely needs to be backported to .NET 7 P1 and so I kept it "minimal".